lizard_connector/query.py

```diff
@@ -0,0 +1,158 @@
+import datetime
+import urllib.parse
+
+from lizard_connector import jsdatetime
+
+
+class LizardApiImproperQueryError(Exception):
+    pass
+
+
+class QueryDictionary(dict):
+
+    def update(self, E=None, *queries, **f):
+        """
+        D.update([E, ]**F) -> None.  Update D from dict/iterable E and F.
+        If E is present and has a .keys() method, then does:  for k in E: D[k] = E[k]
+        If E is present and lacks a .keys() method, then does:  for k, v in E: D[k] = v
+        In either case, this is followed by: for k in F:  D[k] = F[k]
+        """
+        for k, v in f.items():
+            self[k] = v
+        queries = list(queries)
+        if E and isinstance(E, str):
+            for k, v in urllib.parse.parse_qsl(E.strip('?')):
+                self[k] = v
+        elif E:
+            queries.append(E)
+        for d in queries:
+            try:
+                for k, v in d.items():
+                    self[k] = v
+            except AttributeError:
+                raise LizardApiImproperQueryError(
+                    'Query {} is not a dictionary or a string'.format(E))
+
+
+def commaify(*args):
+    """
+    :return: a comma-seperated string of the given arguments
+    """
+    return ','.join(str(x) for x in args)
+
+
+def bbox(south_west, north_east):
+    """
+    Creates a bounding box in Well-known text (WKT).
+    :param south_west: list or tuple with coordinates of the most south western
+                       point of the bounding box of the form (lat, lon)
+    :param north_east: list or tuple with coordinates of the most north eastern
+                       point of the bounding box of the form (lat, lon)
+    :return: bounding box polygon in Well-known text (WKT)
+    """
+    min_lat, min_lon = south_west
+    max_lat, max_lon = north_east
+    polygon_coordinates = [
+            [min_lon, min_lat],
+            [min_lon, max_lat],
+            [max_lon, max_lat],
+            [max_lon, min_lat],
+            [min_lon, min_lat],
+        ]
+    points = [' '.join([str(x), str(y)]) for x, y in polygon_coordinates]
+    return 'POLYGON ((' + ', '.join(points) + '))'
+
+
+def in_bbox(south_west, north_east, endpoint=None):
```

Comment:
Dit zijn allemaal losse functies die je zelf moet samenstellen. Daarvoor is een algemeen querydictsamenstelmechanismeclass-ding gemaakt. En je hebt endpoints waar weinig in gebeurd en die pas nuttig zijn als je op een bepaalde manier queries erin stopt. En een generieke manier om zaken op te halen (verspreidt over twee classes).

Dus... volgens mij vier losse generieke dingen die je alsnog zelf aan elkaar moet plakken. Kan dit comfortabeler?